test/Streamly/Test/Internal/Data/Fold.hs

```diff
@@ -2,26 +2,175 @@ module Main (main) where
 
 import qualified Streamly.Prelude as S
 import Streamly.Internal.Data.Fold
+import qualified Streamly.Data.Fold as FL
 
 import Test.Hspec.QuickCheck
 import Test.QuickCheck (Property, forAll, Gen, vectorOf, arbitrary, choose)
 import Test.QuickCheck.Monadic (monadicIO, assert, run)
-
+import Test.QuickCheck.Arbitrary
 import Test.Hspec as H
 
 maxStreamLen :: Int
 maxStreamLen = 1000
 
+intMin = minBound :: Int
+intMax = maxBound :: Int
+
 testRollingHashFirstN :: Property
-testRollingHashFirstN = 
+testRollingHashFirstN =
     forAll (choose (0, maxStreamLen)) $ \len ->
         forAll (choose (0, len)) $ \n ->
             forAll (vectorOf len (arbitrary :: Gen Int)) $ \vec -> monadicIO $ do
                 a <- run $ S.fold rollingHash $ S.take n $ S.fromList vec
                 b <- run $ S.fold (rollingHashFirstN n) $ S.fromList vec
                 assert $ a == b
 
+testHead :: [Int] -> Expectation
+testHead ls = S.fold FL.head (S.fromList ls) `shouldReturn` headl ls
+            where headl [] = Nothing
```

Comment:
Indentation.

Reply:
https://github.com/composewell/haskell-dev/blob/master/coding-style.rst

But according to this link, a where clause should be in the same column as the `=`
Do you want the `headl` function to start from a new line after the `where`?